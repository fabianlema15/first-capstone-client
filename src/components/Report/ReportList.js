import React from 'react';
import GenericContext from '../../contexts/GenericContext';
import ReportItem from './ReportItem';
import Helper from '../Utils/Helper'

class ReportList extends React.Component{
  static contextType = GenericContext;

  getItems = () => {
    const orderList = this.context.orderList;
    return Object.keys(orderList).map(key =>
      <ReportItem key={orderList[key].id} order={orderList[key]} objLabel={Helper.serializeObj(Helper.getOrdersArray())}/>
    )
  }

  render(){
    return <div>
      {Object.keys(this.context.orderList).length>0 ? this.getItems() : `You have no orders`}
    </div>
  }
}

export default ReportList;
