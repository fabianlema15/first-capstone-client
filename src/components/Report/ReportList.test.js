import React from 'react';
import ReactDOM from 'react-dom';
import ReportList from './ReportList'
import toJson from 'enzyme-to-json'
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('<ReportList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <ReportList>
      </ReportList>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Renders ReportList as expected', () => {
    const context = { objName: 'Test', orderList: {}};
	  const wrapper = shallow(<ReportList/>, { context });
	  expect(toJson(wrapper)).toMatchSnapshot();
	  });

})
