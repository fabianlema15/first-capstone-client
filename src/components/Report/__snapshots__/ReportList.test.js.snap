// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportList /> Renders ReportList as expected 1`] = `
<div>
  You have no orders in this report
</div>
`;
